ansible_base/lib/utils/models.py

```diff
@@ -76,34 +78,111 @@ def user_summary_fields(user):
     return sf
 
 
-def is_system_user(user):
+def is_system_user(user: Optional[AbstractUser]) -> bool:
     """
     Takes a user objects and returns a boolean if that user's username is the same as the SYSTEM_USERNAME
     """
-    setting_name = 'SYSTEM_USERNAME'
-    system_username = get_setting(setting_name)
+    system_username = get_system_username()[0]
     if system_username is None or user is None:
         return False
     return user.username == system_username
 
 
-def get_system_user():
-    system_user = None
-    setting_name = 'SYSTEM_USERNAME'
-    system_username = get_setting(setting_name)
+def get_system_user() -> Optional[AbstractUser]:
+    system_username, setting_name = get_system_username()
     system_user = get_user_model().objects.filter(username=system_username).first()
+    logger.error(f"Got system user {system_user} for {system_username}")
+    # We are using a global variable to try and track if this thread has already spit out the message, if so ignore
+    logger.error(f"{system_username is not None} {system_user is None}")
     if system_username is not None and system_user is None:
+        logger.error("HERE")
         logger.error(
             _(
                 "{setting_name} is set to {system_username} but no user with that username exists.".format(
                     setting_name=setting_name, system_username=system_username
                 )
             )
         )
+        system_user = create_system_user()
+        logger.error(f"Create system user returned {system_user}")
+    logger.error(f"Returning {system_user}")
+    return system_user
+
+
+def get_system_username() -> Tuple[Optional[str], str]:
+    # Returns (system_username, setting_name)
+    setting_name = 'SYSTEM_USERNAME'
+    return get_setting(setting_name), setting_name
+
+
+def create_system_user() -> AbstractUser:
```

Comment:
I think it is a very bad idea to not take `apps` as an input argument here. We need to be much better disciplined about this since we're building a library with DAB and we don't want it to randomly blow up things deep in the migrations spaghetti of some project. Look at the source code for Django `get_user_model`. It is very explicitly clear that it is created with _current_ apps. That means, never call that method in migrations. I think you need to use this in migrations, no?

Reply:
Yea, I think I need to rewrite this function to make it callable from migrations.